python/cudf/cudf/core/dataframe.py

```diff
@@ -446,6 +446,33 @@ def __array_function__(self, func, types, args, kwargs):
     def empty(self):
         return not len(self)
 
+    @property
+    def values(self):
+        if not utils._have_cupy:
+            raise ModuleNotFoundError("cuPY was not found.")
+        import cupy
+
+        return cupy.asarray(self.as_gpu_matrix())
+
+    @property
+    def values_host(self):
+        if self.isnull().all().all():
+            return np.empty(self.shape, dtype=np.object)
+        cols = [
+            c
+            if not is_categorical_dtype(c)
+            else c.as_string_column(dtype=np.object)
+            for c in self._columns
+        ]
+        mat_dtype = np.find_common_type(cols, [])
+        mat = np.zeros(self.shape, dtype=mat_dtype)
+        for i, col in enumerate(cols):
+            if np.issubdtype(col.dtype, np.dtype("object")):
+                mat[:, i] = col.data.to_host()
+            else:
+                mat[:, i] = col.data.mem.copy_to_host()
+        return mat
```

Comment:
Instead of doing all of this, why aren't we just doing `self.as_gpu_matrix().copy_to_host()`?

Reply:
The only reason is that it seemed a bit circuitous to assemble a potentially rather large matrix in device memory when the only purpose is to copy the data to host and then dispose of the gpu matrix.

Reply:
Almost all cuDF operations run out of place and need 2x+ the amount of memory in order to function properly. Doing this on device will be faster and then reduce driver / PCIe pressure by only having a single device --> host memory transfer instead of 1 per column. That sounds like a good tradeoff in my mind.

Reply:
@kkraus14 are we ok with this approach being a little less comprehensive in terms of the cases we can handle? `as_gpu_matrix` has some datatype restrictions (I think due to what numba can and can't handle). Here's a few tests that pop, which I would just mark as xfail: 

`data = {'A': [1, None, 3], 'B': [1.0, 2.0, None]}`
`data = {'A': [1, 2, 3], 'B': ['a', 'b', 'c']}`
`data = {'A': [], 'B': []}`